homeassistant/components/airzone_cloud/binary_sensor.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Final
+
+from aioairzone_cloud.const import AZD_NAME, AZD_PROBLEMS, AZD_WARNINGS, AZD_ZONES
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+from .entity import AirzoneEntity, AirzoneZoneEntity
+
+
+@dataclass
+class AirzoneBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes Airzone Cloud binary sensor entities."""
+
+    attributes: dict[str, str] | None = None
+
+
+ZONE_BINARY_SENSOR_TYPES: Final[tuple[AirzoneBinarySensorEntityDescription, ...]] = (
+    AirzoneBinarySensorEntityDescription(
+        attributes={
+            "warnings": AZD_WARNINGS,
+        },
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        key=AZD_PROBLEMS,
+        name="Problem",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone Cloud binary sensors from a config_entry."""
+    coordinator: AirzoneUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    binary_sensors: list[AirzoneBinarySensor] = []
+
+    for zone_id, zone_data in coordinator.data.get(AZD_ZONES, {}).items():
+        for description in ZONE_BINARY_SENSOR_TYPES:
+            if description.key in zone_data:
+                binary_sensors.append(
+                    AirzoneZoneBinarySensor(
+                        coordinator,
+                        description,
+                        entry,
+                        zone_id,
+                        zone_data,
+                    )
+                )
+
+    async_add_entities(binary_sensors)
+
+
+class AirzoneBinarySensor(AirzoneEntity, BinarySensorEntity):
+    """Define an Airzone Cloud binary sensor."""
+
+    entity_description: AirzoneBinarySensorEntityDescription
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Update attributes when the coordinator updates."""
+        self._async_update_attrs()
+        super()._handle_coordinator_update()
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update binary sensor attributes."""
+        self._attr_is_on = self.get_airzone_value(self.entity_description.key)
+        if self.entity_description.attributes:
+            self._attr_extra_state_attributes = {
+                key: self.get_airzone_value(val)
+                for key, val in self.entity_description.attributes.items()
+            }
```

Comment:
Could use a lambda in the entity description instead

Reply:
I'd rather go this approach since it's the same I used for Airzone (Local), but I'll change it if this change is mandatory:
https://github.com/home-assistant/core/blob/31e9d95f66c3599ce56eceb258d0a48d7bfbb7a1/homeassistant/components/airzone/binary_sensor.py#L129-L136